framework/src/executioners/Executioner.C

```diff
@@ -31,6 +31,12 @@ Executioner::validParams()
   params += Reporter::validParams();
```

Comment:
You possibly want to replace `PicardSolve::validParams()` above with `FixedPointSolve::validParams()`. We assume all three methods `picard`, `secant` and `steffensen` share the same set of valid params thus we can have that `fixed_point_algorithm` parameter.

Reply:
If I do that I think I have to handle all the deprecated Picard parameters in FixedPointSolve. 
We may not assume this in the future. Some methods could have more parameters.